rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/interactor/CovidPatientHistoryActivityInteractorTest.java

```diff
@@ -0,0 +1,51 @@
+    private final int fieldCount = 33;
+    private CovidPatientHistoryActivityInteractor interactor;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        interactor = new CovidPatientHistoryActivityInteractor();
+
+        EventClientRepository repository = new EventClientRepository();
+        Event event = repository.convert(EVENT_JSON, Event.class);
+        EventClient eventClient = new EventClient(event);
+
+        PatientHistoryRepository patientHistoryRepository = Mockito.mock(PatientHistoryRepository.class);
+        ReflectionHelpers.setField(interactor, "patientHistoryRepository", patientHistoryRepository);
+
+        Mockito.doReturn(eventClient).when(patientHistoryRepository).getEvent(ArgumentMatchers.anyString(), ArgumentMatchers.anyString(), ArgumentMatchers.anyString());
+    }
+
+    @Test
+    public void testGetPatientHistoryEntries() throws JSONException {
+        List<PatientHistoryEntry> list = interactor.getPatientHistoryEntries("", "", "");
+
+        Assert.assertNotNull(list);
+        Assert.assertEquals(fieldCount, list.size());
+
+        for (PatientHistoryEntry entry : list) {
+            Assert.assertNotEquals("", entry.getValue());
```

Comment:
Could we verify the actual entry content for a few entries. It may not be enough to check that it's not a blank string.

Reply:
checked first entry